pydoctor/astbuilder.py

```diff
@@ -194,14 +195,58 @@ def extract_final_subscript(annotation: ast.Subscript) -> ast.expr:
         assert isinstance(ann_slice, ast.expr)
         return ann_slice
 
+def getframe(ob:'model.Documentable') -> model.CanContainImportsDocumentable:
+    """
+    Returns the first L{model.Class} or L{model.Module} starting 
+    at C{ob} and going upward in the tree.
+    """
+    if isinstance(ob, model.Inheritable):
+        return getframe(ob.parent)
+    else:
+        assert isinstance(ob, model.CanContainImportsDocumentable)
+        return ob
+
 class ModuleVistor(NodeVisitor):
 
     def __init__(self, builder: 'ASTBuilder', module: model.Module):
         super().__init__()
         self.builder = builder
         self.system = builder.system
         self.module = module
+        self._override_guard_state: Tuple[bool, model.Documentable, Sequence[str]] = \
+                                    (False, cast(model.Documentable, None), [])
+    
+    @contextlib.contextmanager
+    def override_guard(self) -> Iterator[None]:
+        """
+        Returns a context manager that will make the builder ignore any extraneous 
```

Comment:
A clearer explanation of the definition of "extraneous" would be interesting here.

Reply:
extraneous meaning assignments to name that already exist

Reply:
We can read: "Returns a context manager that will make the builder ignore any new assignments"